plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -262,8 +262,21 @@ protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
 	protected function retrieveUserId ($creatorInfo)
 	{
 		$userInfo = $creatorInfo[MicrosoftGraphFieldNames::USER];
-		if(isset($userInfo[MicrosoftGraphFieldNames::EMAIL]))
-		{
+
+		if (isset($userInfo[MicrosoftGraphFieldNames::EMAIL])) {
+			// find user by email and return id if found
+			$filter = new KalturaUserFilter();
+			$filter->emailLike = $userInfo[MicrosoftGraphFieldNames::EMAIL];
+			$response = KBatchBase::$kClient->user->listAction($filter);
+			if ($response->totalCount > 0) {
+				$users = $response->objects;
```

Comment:
@vberezkin-kaltura you could just do a 1 line return here:

return $response->objects[0]->id

Reply:
fixed